src/test/java/com/iexec/sms/tee/session/palaemon/PalaemonSessionServiceTests.java

```diff
@@ -197,8 +207,33 @@ public void shouldGetAppPalaemonTokens() throws Exception {
                 .isEqualTo(Map.of(
                     IexecEnvUtils.IEXEC_INPUT_FILE_NAME_PREFIX + "1", "file1",
                     IexecEnvUtils.IEXEC_INPUT_FILE_NAME_PREFIX + "2", "file2"));
+        assertThat(tokens).containsEntry(APP_PROVIDER_SECRET_0, SECRET_VALUE);
     }
 
+    @Test
+    void shouldGetPalaemonTokensWithoutAppRuntimeSecret() throws Exception {
+        PalaemonSessionRequest request = createSessionRequest();
+        TeeEnclaveConfigurationValidator validator = mock(TeeEnclaveConfigurationValidator.class);
+        final int secretIndex = 0;
+
+        when(enclaveConfig.getValidator()).thenReturn(validator);
+        when(validator.isValid()).thenReturn(true);
+        when(applicationRuntimeSecretService.getSecret(APP_ADDRESS, secretIndex, true))
+                .thenReturn(Optional.empty());
+
+        Map<String, Object> tokens =
+                palaemonSessionService.getAppPalaemonTokens(request);
+        assertThat(tokens).isNotEmpty();
+        assertThat(tokens.get(PalaemonSessionService.APP_MRENCLAVE))
+                .isEqualTo(APP_FINGERPRINT);
+        assertThat(tokens.get(PalaemonSessionService.APP_ARGS))
+                .isEqualTo(APP_ENTRYPOINT + " " + ARGS);
+        assertThat(tokens.get(PalaemonSessionService.INPUT_FILE_NAMES))
+                .isEqualTo(Map.of(
+                        IexecEnvUtils.IEXEC_INPUT_FILE_NAME_PREFIX + "1", "file1",
```

Comment:
We'll need to set constant values instead of things like IexecEnvUtils in UT, but let's stick with that in this PR